girder/web_client/src/views/widgets/ItemListWidget.js

```diff
@@ -140,6 +147,110 @@ var ItemListWidget = View.extend({
+            if (this.tempScrollPos === undefined) {
+                this.tempScrollPos = scrollPos;
+            }
+            // Call a custom scroll event to prevent thinking the user initiated it
+            var e = new CustomEvent('scroll', { detail: 9999 });
+            widgetcontainer[0].scroll(0, scrollPos);
+            widgetcontainer[0].dispatchEvent(e);
+        }
+    },
+    /**
+     * This will look at the position of the selected item and update it as images load and
+     * the DOM reflows
+     */
+    scrollPositionObserver: function () {
+        // Set the default selected height for the selected item
+        const target = $('.g-item-list');
+        if (window.MutationObserver && target.length > 0) {
+            // Default items to monitor for the scroll position
+            const widgetcontainer = $('.g-hierarchy-widget-container');
+            // If the observer already exists disconnect it so it can be recreated
+            if (this.observer && this.observer.disconnect) {
+                this.observer.disconnect();
+            }
+
+            // Event handler for loading images declared once
+            const onLoadImage = (event) => {
+                if (this.observer) {
+                    this.centerSelected();
+                }
+            };
+
+            this.observer = new MutationObserver((mutations) => {
+                // for every mutation
+                _.each(mutations, (mutation) => {
+                    // If no nodes are added we can return
+                    if (!mutation.addedNodes) {
+                        return;
+                    }
+                    // for every added node
+                    _.each(mutation.addedNodes, (node) => {
+                        if (node.className && node.className.indexOf('g-item-list') !== -1) {
+                            // We want to do a onetime scroll to position if the screen is idle
+                            if (window.requestIdleCallback) {
+                                // Processing time is 2 seconds, but should finish much faster
+                                requestIdleCallback(() => {
+                                    this.centerSelected();
+                                }, { timeout: 2000 });
+                            } else {
+                                this.centerSelected();
+                            }
+                        }
+
+                        // For any images added we wait until loaded and rescroll to the selected
+                        if (_.isFunction(node.getElementsByTagName)) {
+                            const imgs = node.getElementsByTagName('img');
+                            for (let i = 0; i < imgs.length; i++) {
+                                // add the event listener to the onload of each image
+                                imgs[i].addEventListener('load', onLoadImage);
+                            }
+                        }
```

Comment:
As long as we are using jQuery, this block could be changed to `$(node).on('load', 'img', onLoadImage)`.  If has the virtue that in new img tags are injected by plugins, they would be watched as well.

Reply:
I've tried to add this but it doesn't seem to call the handler properly.  I could be wrong but for jQuery event delegation to work properly the event has to bubble?   [Jquery .on Documentation](https://api.jquery.com/on/), in the notes at the bottom it specifically talks about how load events don't bubble up.

> In all browsers, the load, scroll, and error events (e.g., on an <img> element) do not bubble. In Internet Explorer 8 and lower, the paste and reset events do not bubble. Such events are not supported for use with delegation, but they can be used when the event handler is directly attached to the element generating the event.
> 
Through the mutation observer all additional img elements should be caught as long as the user doesn't move the scroll area before they are added.  I don't kill the observer until there is user interaction (any scroll not initiated by the program).

Reply:
You're right.  `$('img', node),on('load', onLoadImage)` would work, though that is probably doing exactly what your code is doing anyway.